src/mono/mono/metadata/mono-debug.c

```diff
@@ -1120,6 +1121,24 @@ open_symfile_from_bundle (MonoImage *image)
 {
 	BundledSymfile *bsymfile;
 
+	MonoBundledAssemblyResource *assembly = mono_bundled_resources_get_assembly_resource (image->module_name);
+	if (assembly && assembly->symbol_data.data)
+		return mono_debug_open_image (image, assembly->symbol_data.data, assembly->symbol_data.size);
+
+#ifdef ENABLE_WEBCIL
+	int len = strlen (image->module_name);
+	char *module_name_dll_suffix = strdup (image->module_name);
+	/* if image's module_name ends with .webcil, check if theres a bundled resource with a .dll extension instead */
+	if (module_name_dll_suffix && len >= 7 && !g_strcasecmp (".webcil", &image->module_name [len - 7])) {
+		memcpy (module_name_dll_suffix + len - 7, ".dll", 4);
+		*(module_name_dll_suffix + len - 3) = '\0';
+		assembly = mono_bundled_resources_get_assembly_resource (module_name_dll_suffix);
+	}
+	g_free (module_name_dll_suffix);
+	if (assembly && assembly->symfile)
+		return mono_debug_open_image (image, assembly->symfile.data, assembly->symfile.size);
+#endif
+
 	for (bsymfile = bundled_symfiles; bsymfile; bsymfile = bsymfile->next) {
```

Comment:
Should we do the same to this as with the assembly and satellite assemblies, rewire old API to use the new API?

Reply:
Good catch, forwarded the old bundling api's in mono-debug to leverage the new bundling apis